libcaf_core/test/string_algorithms.cpp

```diff
@@ -0,0 +1,109 @@
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#define CAF_SUITE string_algorithms
+
+#include "caf/string_algorithms.hpp"
+
+#include "caf/test/dsl.hpp"
+
+#include <string>
+#include <vector>
+
+using namespace caf;
+
+namespace {
+
+struct fixture {
+
+};
+
+using str_list = std::vector<std::string>;
+
+str_list split(string_view str) {
+  str_list result;
+  caf::split(result, str, ",");
+  return result;
+}
+
+str_list compressed_split(string_view str) {
+  str_list result;
+  caf::split(result, str, ",", token_compress_on);
+  return result;
+}
+
+std::string join(str_list vec) {
+  return caf::join(vec, ",");
+}
+
+template <class... Ts>
+std::string splice(string_view glue, const Ts&... xs) {
+  std::string result;
+  caf::splice(result, glue, xs...);
+  return result;
+}
+
+} // namespace <anonymous>
+
+CAF_TEST_FIXTURE_SCOPE(string_algorithms_tests, fixture)
+
+CAF_TEST(splitting) {
+  CAF_CHECK_EQUAL(split(""), str_list({""}));
+  CAF_CHECK_EQUAL(split(","), str_list({"", ""}));
+  CAF_CHECK_EQUAL(split(",,"), str_list({"", "", ""}));
+  CAF_CHECK_EQUAL(split(",,,"), str_list({"", "", "", ""}));
+  CAF_CHECK_EQUAL(split("a,b,c"), str_list({"a", "b", "c"}));
+  CAF_CHECK_EQUAL(split("a,,b,c,"), str_list({"a", "", "b", "c", ""}));
+}
+
+CAF_TEST(compressed splitting) {
+  CAF_CHECK_EQUAL(compressed_split(""), str_list({}));
+  CAF_CHECK_EQUAL(compressed_split(","), str_list({}));
+  CAF_CHECK_EQUAL(compressed_split(",,"), str_list({}));
+  CAF_CHECK_EQUAL(compressed_split(",,,"), str_list({}));
+  CAF_CHECK_EQUAL(compressed_split("a,b,c"), str_list({"a", "b", "c"}));
+  CAF_CHECK_EQUAL(compressed_split("a,,b,c,"), str_list({"a", "b", "c"}));
+}
+
+CAF_TEST(joining) {
+  CAF_CHECK_EQUAL(join({}), "");
+  CAF_CHECK_EQUAL(join({""}), "");
+  CAF_CHECK_EQUAL(join({"", ""}), ",");
+  CAF_CHECK_EQUAL(join({"", "", ""}), ",,");
+  CAF_CHECK_EQUAL(join({"a"}), "a");
+  CAF_CHECK_EQUAL(join({"a", "b"}), "a,b");
+  CAF_CHECK_EQUAL(join({"a", "b", "c"}), "a,b,c");
+}
+
+CAF_TEST(starts with) {
+  CAF_CHECK(starts_with("foobar", "f"));
+  CAF_CHECK(starts_with("foobar", "fo"));
+  CAF_CHECK(starts_with("foobar", "fooba"));
+  CAF_CHECK(starts_with("foobar", "foobar"));
+  CAF_CHECK(!starts_with("foobar", "o"));
```

Comment:
Cases where I can easily imagine errors would be something like `starts_with("foobar", "fa")` or `starts_with("foobar", "foobaro")` that include a partial match (same for `ends_with`). Not that I'm expecting this here.

Reply:
Good point. I've added a couple of extra tests.